pynicotine/gtkgui/userbrowse.py

```diff
@@ -688,34 +711,336 @@ def on_select_dir(self, selection):
 
         self.set_directory(iterator.user_data)
 
-    def on_folder_key_press_event(self, *args):
+    def on_select_file(self, selection):
+
+        if selection is None:
+            return
+
+        self.num_selected_files = selection.count_selected_rows()
+
+    def on_key_press_event_folder(self, *args):
 
         keyval, keycode, state, widget = get_key_press_event_args(*args)
-        keycodes, mods = parse_accelerator("Right")
+        path, _focus_column = self.FolderTreeView.get_cursor()
 
-        if keycode in keycodes:
-            path, _focus_column = self.FolderTreeView.get_cursor()
-            self.FolderTreeView.expand_row(path, False)
-            return True
+        _keys, mods_primary = parse_accelerator("<Primary>")
+        _keys, mods_shift = parse_accelerator("<Shift>")
+        _keys, mods_alt = parse_accelerator("<Alt>")
+
+        prs_none = prs_ctrl = prs_shift = prs_alt = False
+
+        if state & mods_primary:  # <GDK_CONTROL_MASK>
+            prs_ctrl = True
+
+        if state & mods_shift:  # <GDK_SHIFT_MASK>
+            prs_shift = True
+
+        if state & mods_alt:  # <GDK_MOD1_MASK>
+            prs_alt = True
+
+        if prs_ctrl == prs_shift == prs_alt is False:
+            prs_none = True  # <GDK_MOD2_MASK> = plain
```

Comment:
These flags are needed to exclude un-pressed modifier keys from the bindings, since parse_accelerator doesn't seem to be capable of avoiding conflicts by itself. Since this needs to happen for every on_key_press_event function, I would like to do this externally like in utils.py, but I don't yet know how to best accomplish this.

Reply:
See https://github.com/nicotine-plus/nicotine-plus/pull/1587/files#r720785164